packages/cardano/src/cardano/api/token-supply.ts

```diff
@@ -0,0 +1,66 @@
+import {z} from 'zod'
+import {createTypeGuardFromSchema, fetcher} from '@yoroi/common'
+
+import {
+  ApiTokenId,
+  ApiTokenSupplyRecord,
+  ApiTokenSupplyResponse,
+  ApiTokeSupplyRequest,
+} from './types'
+import {getTokenIdentity} from '../translators/formatters/cardano-token-id'
+
+export const getTokenSupply =
+  (baseUrl: string, request = fetcher) =>
+  async (tokenIds: ApiTokeSupplyRequest): Promise<ApiTokenSupplyResponse> => {
+    if (tokenIds.length === 0) {
+      return Promise.resolve({})
+    }
+
+    const assetsMap = new Map<string, {policy: string; name: string}>(
+      tokenIds.map((id) => {
+        const {policyId: policy, name} = getTokenIdentity(id)
+        return [id, {policy, name}]
+      }),
+    )
+
+    const payload = {assets: Array.from(assetsMap.values())}
+
+    return request<ApiTokenSupplyResponse>({
+      url: `${baseUrl}/multiAsset/supply`,
+      data: payload,
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+    }).then((response) => {
+      const parsedResponse = parseTokenSupplyResponse(response)
+      if (!parsedResponse)
+        return Promise.reject(new Error('Invalid asset supplies'))
+
+      const result: any = {} // need any here TS issues with key indexing
```

Comment:
same issue when you don't have explicit keys

Reply:
the expected output is `Record<ApiTokenId, ApiTokenSupplyRecord>` if we use `Record<string...` it will work for the 1st complain, but is not the same type of the return `Record<ApiTokenId..` no idea why TS can't handle typed keys.

Reply:
I've found the issue, the mapping was declared as `string`, and not as `ApiTokenId`.